Watchman.Discord/Areas/Users/Controllers/WarnsController.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Framework.Architecture.Controllers;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Framework.Commands.Responses;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Factories;
+using MongoDB.Bson;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Discord.Areas.Protection.Services;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.Discord.Areas.Users.BotCommands.Warns;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Users;
+
+namespace Watchman.Discord.Areas.Protection.Controllers
+{
+    public class WarnsController : IController
+    {
+        private readonly MessagesServiceFactory _messagesServiceFactory;
+        private readonly UsersService _usersService;
+        private readonly DirectMessagesService _directMessagesService;
+        private readonly WarnsService _warnService;
+
+        public WarnsController(MessagesServiceFactory messagesServiceFactory, UsersService usersService, DirectMessagesService directMessagesService, WarnsService warnService)
+        {
+            this._messagesServiceFactory = messagesServiceFactory;
+            this._usersService = usersService;
+            this._directMessagesService = directMessagesService;
+            this._warnService = warnService;
+        }
+
+        [AdminCommand]
+        public async Task AddWarn(AddWarnCommand command, Contexts contexts)
+        {
+            var messageService = this._messagesServiceFactory.Create(contexts);
+            var mentionedUser = await this._usersService.GetUserByMentionAsync(contexts.Server, command.User.ToString());
+            await this._warnService.AddWarnToUser(command, contexts, mentionedUser);
+            await messageService.SendResponse(x => x.UserHasBeenWarned(contexts.User.Name, mentionedUser.Name, command.Reason));
+        }
+
+        public async Task Warns(WarnsCommand command, Contexts contexts)
+        {
+            var mentionedUser = command.User == 0 ? contexts.User : await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
+            var serverId = command.All ? 0 : contexts.Server.Id;
+            await (command.All ? this._warnService.GetAllWarns(command, contexts, mentionedUser, serverId)
+                : this._warnService.GetWarns(command, contexts, mentionedUser, serverId));
```

Comment:
te skrócone warunki w tym miejscu wydają się zbyt skomplikowane 
zrobiłbym normalny warunek z klamrami np. `if (command.All)`

Reply:
druga rzecz, to metoda GetWarns jak nazwa wskazuje powinna zwracać warny, a tutaj zaskoczeniem jest że je wyświetla
lepiej zwracaj warny i wyświetleniem ich zajmij się w kontrolerze (bo to nie jest jakieś długie i skomplikowane, więc może być tutaj)

Reply:
ewentualnie jeśli jednak kod byłby zbyt długi jak na kontroler to zmień nazwę/dodaj nową metodę w tamtym serwisie z nazwą PrintWarns/PrintAllWarns

Reply:
miałem już if (command.All) ale zmieniłem na prośbę kogoś (chyba Marcina) w review

Reply:
już nie ma -all więc nieaktualne